site/app/templates/course/CourseMaterials.twig

```diff
@@ -334,6 +334,14 @@
                             switch (index) {
                                 case 0: 
                                     date = new Date();
+                                    var offset = date.getTimezoneOffset();
+                                    offset = offset/60;
+                                    console.log(offset);
+                                    //if your time is ahead of New York, sets back a day so
+                                    //everything is simultaneous
+                                    if(offset<4){
+                                       date.setUTCDate(date.getDate()-1);
+                                    }
```

Comment:
I dont think moving the day back is the best way to do this because it is not always the case that the server's timezone is EST in which case your offset would be wrong. I suggest taking the user's time no matter where they are and converting it to UTC and do all you calculation on when to display it with that time. Then when you display it convert it back to the user's time.

Reply:
I agree with this, I am also a firm believer that all timestamps should be stored in UTC using [atom datetime format](https://www.php.net/manual/en/class.datetimeinterface.php#datetime.constants.atom).

Reply:
Per @bmcutler, all dates are stored with their respective timezones (and written to json files and log files) to make it easier to debug time-related bugs and issues so that instructors and newbie sysadmins don't have to worry about constantly doing timezone conversions on dates to their local time.

Reply:
ok I'll take saketh's/kevin/matt's suggestions

Reply:
You could have a look at how it stored timestamp before your edit. 

My `course_materials_file_data.json` contains some old data and they look like this:

```
    "/var/local/submitty/courses/s19/sample/uploads/course_materials/TicTacToe-orig/.settings/org.eclipse.jdt.core.prefs": {
        "checked": "1",
        "release_datetime": "2019-06-03 08:10:13-0400"
    },
    "/var/local/submitty/courses/s19/sample/uploads/course_materials/TicTacToe-orig/gen/org/example/tictactoe/Manifest.java": {
        "checked": "1",
        "release_datetime": "2019-06-03 08:10:20-0400"
    },
    "/var/local/submitty/courses/s19/sample/uploads/course_materials/TicTacToe-orig/bin/resources.ap_": {
        "checked": "1",
        "release_datetime": "2019-06-03 08:40:30-0400"
    },
```

but after changing to the new course material interface the json now stores things like this:

```
    "/var/local/submitty/courses/s19/sample/uploads/course_materials/Lab1/Lab01.pdf": {
        "checked": "1",
        "release_datetime": "2019-07-06 12:00:00"
    },
    "/var/local/submitty/courses/s19/sample/uploads/course_materials/Lab1/Lab1.c": {
        "checked": "1",
        "release_datetime": "2019-07-06 12:00:00"
    },
```